client/dashboard/app/snackbars/index.tsx

```diff
@@ -13,9 +17,46 @@ const statusIcon: Record< string, React.JSX.Element > = {
 	error,
 };
 
+declare module '@tanstack/react-query' {
+	interface Register {
+		mutationMeta: {
+			snackbar?: {
+				success?: string;
+				error?: string;
+			};
+		};
+	}
+}
+
 export default function Snackbars() {
 	const notices = useSelect( ( select ) => select( noticesStore ).getNotices(), [] );
-	const { removeNotice } = useDispatch( noticesStore );
+	const { removeNotice, createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	// Displays snackbars which have been requested through the `meta` option of
+	// the `useMutation` hook.
+	useEffect( () => {
+		const cleanupSuccess = registerMutationSuccessCallback(
```

Comment:
I guess we can use the [mutationCache.subscribe](https://tanstack.com/query/latest/docs/reference/MutationCache#mutationcachesubscribe) function to handle the success/error. If so, we don't need to create `registerMutationSuccessCallback`/`registerMutationErrorCallback` manually.

```js
return queryClient.getMutationCache().subscribe( ( { type, mutation } ) => {
  const createNotice = type === 'success' ? createSuccessNotice : createErrorNotice;
  const message = mutation.meta?.snackbar?.[ type ];
  if ( message ) {
    createNotice( message, { type: 'snackbar' } );
  }
});
```

Reply:
@arthur791004 done in 30c633badb3651362a6af8132db18c258e6d937e

It took a while to find the equivalent event parameters to test for. The trick is to look for `type==='updated'`, and when we know it's an update we're allowed to access the `event.action`.